lstchain/reco/reconstructor.py

```diff
@@ -0,0 +1,951 @@
+
+        Returns
+        -------
+        cam_display: `ctapipe.visualization.CameraDisplay`
+            Camera image using matplotlib
+
+        """
+
+        params = self.end_parameters
+
+        rl = 1 + params['rl'] if params['rl'] >= 0 else 1 / (1 - params['rl'])
+        mu = asygaussian2d(np.float32(params['charge']
+                                      * self.geometry.pix_area.to_value(u.m ** 2)),
+                           np.float32(self.geometry.pix_x.value),
+                           np.float32(self.geometry.pix_y.value),
+                           np.float32(params['x_cm']),
+                           np.float32(params['y_cm']),
+                           np.float32(params['wl'] * params['length']),
+                           np.float32(params['length']),
+                           np.float32(params['psi']),
+                           np.float32(rl))
+        residual = image - mu
+
+        cam_display = display_array_camera(residual,
+                                           camera_geometry=self.geometry)
+        if save:
+            cam_display.axes.get_figure().savefig('event/' + ids +
+                                                  '_residuals.png')
+            plt.close()
+        return None if save else cam_display
+
+    def plot_model(self, save=False, ids=''):
+        """
+        Create a CameraDisplay object showing the spatial model fitted to
+        the current event
+
+        Parameters
+        -------
+        save: bool
+            Save and close the figure if True, return it otherwise
+        ids: string
+            Can be used to modify the save location
+
+        Returns
+        -------
+        cam_display: `ctapipe.visualization.CameraDisplay`
+            Camera image using matplotlib
+
+        """
+
+        params = self.end_parameters
+        rl = 1 + params['rl'] if params['rl'] >= 0 else 1 / (1 - params['rl'])
+        mu = asygaussian2d(np.float32(params['charge']
+                                      * self.geometry.pix_area.to_value(u.m ** 2)),
+                           np.float32(self.geometry.pix_x.value),
+                           np.float32(self.geometry.pix_y.value),
+                           np.float32(params['x_cm']),
+                           np.float32(params['y_cm']),
+                           np.float32(params['wl'] * params['length']),
+                           np.float32(params['length']),
+                           np.float32(params['psi']),
+                           np.float32(rl))
+
+        cam_display = display_array_camera(mu,
+                                           camera_geometry=self.geometry)
+        if save:
+            cam_display.axes.get_figure().savefig('event/' + ids +
+                                                  '_model.png')
+            plt.close()
+        return None if save else cam_display
+
+    def plot_waveforms(self, image, axes=None, save=False, ids=''):
```

Comment:
all the plotting code should go in different classes, probably into `visualization`

Reply:
It is now the case